src/state/mod.rs

```diff
@@ -1,34 +1,38 @@
-use ggez::event;
+use ggez::event::{EventHandler, KeyCode, KeyMods};
 use ggez::graphics;
 use ggez::nalgebra as na;
 use ggez::{Context, GameResult};
 
 use crate::draw::clear_background;
 use crate::ball::get_ball_graphics;
 use crate::paddle::get_paddle_graphics;
+use crate::player::{PADDLE_WIDTH, PADDLE_HEIGHT, Player};
 
-const PADDLE_WIDTH : f32 = 30.0;
-const PADDLE_HEIGHT: f32 = 150.0;
+pub const WINDOW_WIDTH: f32 = 800.0;
+pub const WINDOW_HEIGHT: f32 = 600.0;
 
 pub struct MainState {
     pub ball_position: (f32, f32),
-    player_position: (f32,f32),
-    enemy_position: (f32, f32),
+    player1: Player,
+    player2: Player,
 }
 
 impl MainState {
     pub fn new() -> GameResult<MainState> {
         let s = MainState {
             ball_position: (400.0, 300.0),
-            player_position: (40.0, 225.0),
-            enemy_position: (800.0 - 70.0, 225.0),
+            player1: Player::new(40.0, 225.0),
+            player2: Player::new(800.0 - 70.0, 225.0),
```

Comment:
why `800 - 70`? maybe `WINDOW_WIDTH - 70`

Reply:
40 could be something like `GOAL_DISTANCE`

Reply:
Nice! Fixed this one too :slightly_smiling_face: